pkg/apis/gcp/cache/v1alpha1/register.go

```diff
@@ -45,9 +39,12 @@ var (
 
 	// SchemeBuilder is used to add go types to the GroupVersionKind scheme
 	SchemeBuilder = &scheme.Builder{GroupVersion: SchemeGroupVersion}
+)
 
-	// CloudMemorystoreInstanceGroupVersionKind is the GVK of a
-	// CloudMemorystoreInstance.
+// CloudMemorystoreInstance type metadata.
+var (
+	CloudMemorystoreInstanceKind             = reflect.TypeOf(CloudMemorystoreInstance{}).Name()
```

Comment:
Yeah - I changed to this method because lookups in the scheme (i.e. to get a `runtime.Object` given its GVK) require an uppercase kind. This reflect.TypeOf dance is pretty much what the scheme does under the hood. I believe I addressed the provider case issue by doing a case insensitive compare elsewhere in this PR.